### PR TITLE
Expose string comparison string.Replace overloads

### PR DESCRIPTION
**Pull request by hughbe**
Fixes https://github.com/dotnet/corefx/issues/1271

Currently has some disabled tests due to bugs that I'm planning to fix in coreclr

### CODE REVIEW

File: src/System.Runtime/ref/System.Runtime.cs [hunk 1]
```diff
@@ -5,6 +5,8 @@
 // Changes to this file must follow the http://aka.ms/api-review process.
 // ------------------------------------------------------------------------------
 
+using System.Globalization;
```
Comment (stephentoub) [inline src/System.Runtime/ref/System.Runtime.cs]: We don't do this in our ref files.  Instead use the namespace-qualified name on every required usage.

File: src/System.Runtime/tests/System/StringTests.netcoreapp.cs [hunk 2]
```diff
@@ -123,5 +124,156 @@ public static void Join_Char_InvalidStartIndexCount_ThrowsArgumentOutOfRangeExce
...
+        }
+
+        [Fact]
+        public void Replace_StringComparsion_NullOldValue_ThrowsArgumentException()
+        {
+            Assert.Throws<ArgumentNullException>("oldValue", () => "abc".Replace(null, "def", StringComparison.CurrentCulture));
+            Assert.Throws<ArgumentNullException>("oldValue", () => "abc".Replace(null, "def", true, CultureInfo.CurrentCulture));
+        }
+
+        //TODO: [Fact]
```
Comment (stephentoub) [inline src/System.Runtime/tests/System/StringTests.netcoreapp.cs]: ?

**Comment by stephentoub** [comment]
Thanks for adding all of these, @hughbe.

File: src/System.Runtime/tests/System/StringTests.netcoreapp.cs (hunk 2 above)
Comment (hughbe) [inline src/System.Runtime/tests/System/StringTests.netcoreapp.cs]: This currently fails with an OOM exception: https://github.com/dotnet/coreclr/pull/11001

Comment (stephentoub) [inline src/System.Runtime/tests/System/StringTests.netcoreapp.cs]: Ok, then we should have:
```C#
[ActiveIssue("https://github.com/dotnet/coreclr/pull/11001")]
[Fact]
```

**Comment by stephentoub** [comment]
@dotnet-bot test Innerloop CentOS7.1 Release x64 Build and Test please (https://github.com/dotnet/corefx/issues/18431)

**Comment by tarekgh** [comment]
@hughbe thank you!

**Pull request merged**